### PR TITLE
80 clean up left over job status progress and result queues

### PR DESCRIPTION
**Pull request by cwang39403**
Related to https://github.com/Project-OMOTES/omotes-system/issues/80#issue-2405856101

Dependent on the open PR #60
and on SDK https://github.com/Project-OMOTES/omotes-sdk-python/pull/63

### CODE REVIEW

**Comment by lfse-slafleur** [comment]
Tip: As you have another branch merged in, you can also set the target to the latest branch that is merged into this branch. This will tell Github to only show the changes this branch introduces as well as tell us that we should first merge another PR instead of this one. I have changed the target to show this.

**Comment by lfse-slafleur** [comment]
Oh and P.S. if that target branch is deleted, it automatically shifts the target back to main so we do not have to change the target back to main ourselves anymore.

File: src/omotes_orchestrator/main.py [hunk 1]
```diff
@@ -297,10 +302,20 @@ def new_job_submitted_handler(self, job_submission: JobSubmission) -> None:
             )
         else:
             logger.debug("New job %s was not yet registered. Registering and submitting.")
+
+            if not job_submission.HasField("timeout_ms"):
```
Comment (lfse-slafleur) [inline src/omotes_orchestrator/main.py:306]: I am very confused as to why these changes are showing. This was added in commit https://github.com/Project-OMOTES/orchestrator/commit/3e10ae836ffe4f763af73838672a3263a6728c29 on the 30th of July and this commit is NOT in the PR. It appears to be a Github bug. Perhaps merging main into this branch helps as I have seen this before and that usually fixes it.

Comment (lfse-slafleur) [reply src/omotes_orchestrator/main.py]: Yup, merging main into this branch fixed it again. Appears to be an issue where Github cannot figure out the merge apriory properly. Git used an ORT strategy so it wasn't just a simple fastforward.

File: src/omotes_orchestrator/main.py [hunk 2]
```diff
@@ -396,6 +401,40 @@ def job_cancellation_handler(self, job_cancellation: JobCancel) -> None:
...
+        to a local file.
+
+        :param job_result: Job result message.
+        """
+        logger.info("Received a dead lettered job_%s_result with result type as: %s",
+                    job_result.uuid,
+                    job_result.result_type)
+
+        log_level = logger.getEffectiveLevel()
+        if log_level == logging.DEBUG:
```
Comment (lfse-slafleur) [inline src/omotes_orchestrator/main.py:417]: Log level may also be `NOT_SET`. We should check `log_level <= logging.DEBUG` (https://docs.python.org/3/library/logging.html#logging-levels)

File: src/omotes_orchestrator/main.py [hunk 3]
```diff
@@ -396,6 +401,40 @@ def job_cancellation_handler(self, job_cancellation: JobCancel) -> None:
...
+            try:
+                log_dir = "../logs"
```
Comment (lfse-slafleur) [inline src/omotes_orchestrator/main.py:419]: This path should be configurable

File: src/omotes_orchestrator/main.py [hunk 4]
```diff
@@ -396,6 +401,40 @@ def job_cancellation_handler(self, job_cancellation: JobCancel) -> None:
...
+                with open(file_path, "a") as f:
+                    f.write("--------------Logs:\n")
+                    f.write(job_result.logs + "\n")
+                    f.write("--------------Status:\n")
+                    f.write(str(job_result.result_type) + "\n")
+                    f.write("--------------ESDL:\n")
+                    f.write(job_result.output_esdl + "\n")
+
+                logger.info("The job result is logged to: %s", file_path)
+            except Exception as e:
```
Comment (lfse-slafleur) [inline src/omotes_orchestrator/main.py:435]: Good idea! Lets not let this best-effort logging break the application.

**Review by lfse-slafleur — CHANGES_REQUESTED** [review]

**Review by MarkTNO — APPROVED** [review]

File: src/omotes_orchestrator/main.py (hunk 2 above)
Comment (cwang39403) [reply src/omotes_orchestrator/main.py]: Good tips! Updated in 1a5bb3fe40a03f272badacea7bd8e413a3c8c4a5

File: src/omotes_orchestrator/main.py (hunk 3 above)
Comment (cwang39403) [reply src/omotes_orchestrator/main.py]: Updated in 1a5bb3fe40a03f272badacea7bd8e413a3c8c4a5

I now make it as an Orchestrator class attribute as I don't feel it is needed to set this in `.env` file. If there is a better way of doing it, let me know!

File: src/omotes_orchestrator/main.py (hunk 1 above)
Comment (cwang39403) [reply src/omotes_orchestrator/main.py]: Thanks for checking this. I wasn't aware that git automatically shifts the target branch back to the main when the original target branch is deleted. A good tip!

**Comment by lfse-slafleur** [comment]
@cwang39403 SDK 3.1.0 has just been released and should contain the code changes to fix this PR's checks.

**Comment by lfse-slafleur** [comment]
@cwang39403 Hope you do not mind I added a commit to solely update the SDK. Considering it was such a small change, I thought it would be more practical to help out. Reviewing now!

**Review by lfse-slafleur — APPROVED** [review]

**Comment by cwang39403** [comment]
@lfse-slafleur Thanks for updating the SDK, I am about to work on it this afternoon, but it is even better that you made this change already. I will merge the PR.
